mycore-mods/src/main/java/org/mycore/mods/classification/MCRClassificationMappingEventHandler.java

```diff
@@ -151,6 +157,45 @@ private static String getXPathMappingGenerator(MCRCategoryID target) {
             GENERATOR_SUFFIX);
     }
 
+    /**
+     * Replaces a specific pattern in an XPath with the value of a matching property. Placeholders in the
+     * property value are substituted with the specific values given in an XPath.<p>
+     * Syntax:<p>
+     * {pattern:&lt;name of property&gt;=&lt;comma-separated list of values&gt;}<p>
+     * (when there are no values, the "=" is optional)<p>
+     * Ex.:<p>
+     * <b>Input XPath:</b> {pattern:Genre=article} and not(mods:relatedItem[@type='host'])<p>
+     * <b>Property:</b> MCR.Category.XPathMapping.Pattern.Genre=mods:genre[substring-after(@valueURI,'#')='{0}']<p>
+     * <b>Substituted XPath:</b> mods:genre[substring-after(@valueURI,'#')='article']
+     * and not(mods:relatedItem[@type='host'])
+     * @param xPath the XPath containing a pattern to substitute
+     * @return the resolved xPath
+     */
+    private static String replacePattern(String xPath) {
```

Comment:
1) With an improved Regex (more groups) you could further reduce the String operations:
https://regex101.com/r/7H2fcJ/3

2) You should support multiple occurrences: `while (matcher.finde())` should do the work

3) You may use the string positions provided by the regex matcher:
 `xpath = xpath.substring(0, matcher.start()) + placeholderText + xpath.substring(matcher.end())`